### PR TITLE
Added IT tests for RFC 7232 for Directory Service

### PR DESCRIPTION
**Pull request by sanjay-saxena**

### CODE REVIEW

File: service/http.directory/src/test/java/org/kaazing/gateway/service/http/directory/rfc7232/IfMatchIT.java [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+import org.kaazing.gateway.server.test.GatewayRule;
+import org.kaazing.gateway.server.test.config.GatewayConfiguration;
+import org.kaazing.gateway.server.test.config.builder.GatewayConfigurationBuilder;
+
+/**
+ * RFC-7232, section 3.1 "If-Match"
+ */
+public class IfMatchIT {
+    private final K3poRule k3po = new K3poRule().setScriptRoot("org/kaazing/specification/http/rfc7232/preconditions/if.match");
+    private final TestRule timeout = new DisableOnDebug(new Timeout(5, SECONDS));
```
Comment (dpwspoon) [inline service/http.directory/src/test/java/org/kaazing/gateway/service/http/directory/rfc7232/IfMatchIT.java]: See ITUtilRule, I have no preference on what we use here but I think Chris is trying to get people to standardize

File: pom.xml [hunk 2]
```diff
@@ -356,6 +350,38 @@
                 </configuration>
             </plugin>
             <plugin>
```
Comment (dpwspoon) [inline pom.xml]: Thanks for fixing this

File: service/http.directory/src/test/java/org/kaazing/gateway/service/http/directory/rfc7232/IfMatchIT.java (hunk 1 above)
Comment (sanjay-saxena) [inline service/http.directory/src/test/java/org/kaazing/gateway/service/http/directory/rfc7232/IfMatchIT.java]: I think John wants the tests to be in terms of primitives. So, once we have a resolution, we can go and change all the tests to conform to one directive or the other.

**Comment by dpwspoon** [comment]
Looks like you have a build error that needs to be resolved

```
'dependencies.dependency.version' for org.kaazing:specification.http:jar is missing. 
```

**Pull request merged**